### DISCUSSION TITLE
Can't wake up the player from standby mode

### DISCUSSION THREAD

**Discussion by PowercoderJr** [Makuna/DFMiniMp3#90] [Q&A] [answered]
Did anyone face the problem of getting the player unresponsive after using `sleep()` method? Neither `reset()` nor `setPlaybackSource()` nor `0x0B` command (why is there no method for it in the lib btw?) helps. I can hear my speaker stops emitting silent noise after `sleep()` and no reaction on any command since. `onError()` is not triggered too.

**Comment by seisfeld** [comment, marked as answer]
Yes. The problem ist just there's no "the player". There are over half a dozen (if not more) different chips used on these modules. All with potentially different firmwares (even if it's the same chip). Hence you can't generalize the functionality, it's always hit and miss and trial and error. Especially recently where the "good" chips are not produced anymore and Chinese vendors have to switch and apparently have to re-do the firmwares.

**Reply by PowercoderJr** [reply]
So does it mean I can't use standby mode or I need to figure out how to use it somehow? [Here is mine](https://aliexpress.ru/item/1005003870017446.html).

**Comment by seisfeld** [comment]
Well, judging from the image - if you really have the MH2024K-24SS - you IMHO have one of the better ones. But yes, if the mentioned functions don't work on your unit you are kinda out of luck. I don't know im what scenario you use your player but maybe cutting the power to your setup completely is also an option. A MCU starts up pretty quick, so maybe you don't really "need" putting the player to sleep.

**Reply by PowercoderJr** [reply]
Kinda offtop: I use a [boost converter](https://aliexpress.ru/item/32820218683.html) just after batteries and its output is connected to a player. Is it actually useful to switch to standby mode or energy consumption will be the same?

**Reply by PowercoderJr** [reply]
There is one thing, maybe it is some kind of built-in energy saving mode? When player is idle it needs about 0.5-1 sec after sending play command to actually emit the sound. If I send it while previous track is sounding, the next starts immediately. Have you ever noticed this?

**Comment by Makuna** [comment]
What is 0x0b supposed to be?  My spec sheet just lists it as reserved.

You can try other commands (like play track, etc) and check if the status changes `getStatus()` which is also a weird call in what it returns and where it is supported.

You might try calling sleep again, to see if wakes up on the second time.

**Reply by PowercoderJr** [reply]
> What is 0x0b supposed to be? My spec sheet just lists it as reserved.

It states as "normal working" [here](http://www.electronicoscaldas.com/datasheet/DFR0299-DFPlayer-Mini-Manual.pdf). Can you try it?

> You can try other commands

No results. Status request returns error 129 but it worked before standby.

> You might try calling sleep again, to see if wakes up on the second time.

Still.

**Reply by Makuna** [reply]
https://github.com/Makuna/DFMiniMp3/pull/91
I added the awake command (might rename it before releasing an official build with it).
